src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx

```diff
@@ -26,35 +40,53 @@ export function LimitOrdersWidget() {
   }
   const inputCurrencyInfo: CurrencyInfo = {
     field: Field.INPUT,
-    currency: USDC[1],
-    rawAmount: null,
-    viewAmount: '1',
+    currency: inputCurrency,
+    rawAmount: inputCurrencyAmount,
+    viewAmount: inputCurrencyAmount?.toExact() || '',
     balance: null,
     fiatAmount: null,
     receiveAmountInfo: null,
   }
   const outputCurrencyInfo: CurrencyInfo = {
-    field: Field.INPUT,
-    currency: GNO[1],
-    rawAmount: null,
-    viewAmount: '2',
+    field: Field.OUTPUT,
+    currency: outputCurrency,
+    rawAmount: outputCurrencyAmount,
+    viewAmount: outputCurrencyAmount?.toExact() || '',
     balance: null,
     fiatAmount: null,
     receiveAmountInfo: null,
   }
-  const onCurrencySelection = () => {
-    //
-  }
-  const onUserInput = () => {
-    //
-  }
+  const onCurrencySelection = useCallback(
+    (field: Field, currency: Currency) => {
+      const inputCurrencyId = (field === Field.INPUT ? currency.symbol : inputCurrency?.symbol) || ''
```

Comment:
Nitpick. I think we could make this code simpler to read.

Given that you only want to set `inputCurrencyId` and `outputCurrencyId` depending on the `Field` enum, which only has two values. Why not using a simple `if/else` or a `Switch` this way reads closer to natutal language

Reply:
Fixed: https://github.com/cowprotocol/cowswap/pull/1110/commits/ee8909f7dacf04c7460fb5413b16ccedb532ed54